ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestconnectivity.py

```diff
@@ -177,18 +176,18 @@ def test_definition(self) -> dict | None:
         """
         anta_tests = []
 
-        if (ethernet_interfaces := self.logged_get(key="ethernet_interfaces", logging_level="WARNING")) is None:
+        if (ethernet_interfaces := self.structured_config.get("ethernet_interfaces")) is None:
+            LOGGER.warning("No ethernet interfaces found. %s is skipped.", self.__class__.__name__)
             return None
 
         required_keys = ["name", "peer", "peer_interface"]
 
         for idx, interface in enumerate(ethernet_interfaces):
-            # ignore subinterfaces for LLDP - TODO makes this better
-            if self.is_subinterface(interface):
+            self.update_interface_shutdown(interface)
+            if not self.validate_data(data=interface, data_path=f"ethernet_interfaces.[{idx}]", required_keys=required_keys, shutdown=False):
                 continue
 
-            self.update_interface_shutdown(interface=interface)
-            if not self.validate_data(data=interface, data_path=f"ethernet_interfaces.[{idx}]", required_keys=required_keys, shutdown=False):
+            if self.is_subinterface(interface):
```

Comment:
why change the order? now we are "wasting" cycle updat8ing subinterface?

Reply:
Moved and added logs in 554e20b10cc2d157c35c92107939d8e54417c893